File: action_demo/src/dishes_Server.cpp
   #include <action_demo/DoDishesAction.h>  // Note: "Action" is appended
   #include <actionlib/server/simple_action_server.h>

   typedef actionlib::SimpleActionServer<action_demo::DoDishesAction> Server;

   void execute(const action_demo::DoDishesGoalConstPtr& goal, Server* as)  // Note: "Action" is not appended to DoDishes here
   {
      // Do lots of awesome groundbreaking robot stuff here
     as->setSucceeded();
   }

  int main(int argc, char** argv)
   {
     ros::init(argc, argv, "dishes_Server");
    ros::NodeHandle n;
     Server server(n, "dishes", boost::bind(&execute, _1, &server), false);
    server.start();
     ros::spin();
    return 0;
   }

File: action_demo/src/dishes_Server.py
  #! /usr/bin/env python

  import roslib
  roslib.load_manifest('my_pkg_name')
  import rospy
  import actionlib

  from action_demo.msg import DoDishesAction

  class DoDishesServer:
    def __init__(self):
      self.server = actionlib.SimpleActionServer('dishes', DoDishesAction, self.execute, False)
      self.server.start()

    def execute(self, goal):
      # Do lots of awesome groundbreaking robot stuff here
      self.server.set_succeeded()


  if __name__ == '__main__':
    rospy.init_node('dishes_server')
   server = DoDishesServer()
   rospy.spin()
